src/main/java/uk/gov/hmcts/darts/arm/service/impl/DetsToArmBatchPushProcessorImpl.java

```diff
@@ -0,0 +1,340 @@
+            }
+        } catch (Exception e) {
+            String errorMessage = String.format("Error during generation of DETS batch manifest file %s", archiveRecordsFile.getName());
+            log.error(errorMessage, e);
+            batchItems.getSuccessful().forEach(batchItem -> dataStoreToArmHelper.recoverByUpdatingEodToFailedArmStatus(batchItem, userAccount));
+            final String errorMessageWithStackTrace = errorMessage + " - " + ExceptionUtils.getStackTrace(e);
+            batchItems.getFailed().forEach(batchItem -> updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessageWithStackTrace));
+            return false;
+        }
+        return true;
+    }
+
+    private String getManifestFilePrefix() {
+        return detsToArmProcessorConfiguration.getManifestFilePrefix();
+    }
+
+    private void updateObjectStateRecordManifestSuccessOrFailure(ArmBatchItems batchItems, File archiveRecordsFile) {
+        for (var batchItem : batchItems.getSuccessful()) {
+            ObjectStateRecordEntity objectStateRecord = getObjectStateRecordEntity(batchItem.getArmEod());
+            if (nonNull(objectStateRecord)) {
+                objectStateRecord.setFlagFileMfstCreated(true);
+                objectStateRecord.setDateFileMfstCreated(currentTimeHelper.currentOffsetDateTime());
+                objectStateRecord.setIdManifestFile(archiveRecordsFile.getName());
+                objectStateRecord.setFlagMfstTransfToArml(true);
+                objectStateRecord.setDateMfstTransfToArml(currentTimeHelper.currentOffsetDateTime());
+                objectStateRecordRepository.save(objectStateRecord);
+            }
+
+        }
+        for (var batchItem : batchItems.getFailed()) {
+            String errorMessage = "Manifest file creation failed";
+            updateObjectStateRecordStatus(batchItem.getArmEod(), errorMessage);
+        }
+
+    }
+
+    private void updateObjectStateRecordStatus(ExternalObjectDirectoryEntity externalObjectDirectory, String errorMessage) {
+        ObjectStateRecordEntity objectStateRecord = getObjectStateRecordEntity(externalObjectDirectory);
+        if (nonNull(objectStateRecord)) {
+            if (nonNull(objectStateRecord.getObjectStatus())) {
+                objectStateRecord.setObjectStatus(objectStateRecord.getObjectStatus() + " " + errorMessage);
+            } else {
+                objectStateRecord.setObjectStatus(errorMessage);
+            }
+            objectStateRecordRepository.save(objectStateRecord);
+        } else {
+            if (nonNull(externalObjectDirectory)) {
+                log.error("Unable to find ObjectStateRecordEntity for ARM EOD ID: {}", externalObjectDirectory.getId());
+            }
+        }
+    }
+
+    private ObjectStateRecordEntity getObjectStateRecordEntity(ExternalObjectDirectoryEntity externalObjectDirectory) {
+        if (nonNull(externalObjectDirectory.getOsrUuid())) {
```

Comment:
I don't think this is necessary, we can never be in a situation where this field is `null` because at the very beginning the processor has only retrieved from the DB EODs with ors uuid not null. 
I would remove it since it complicates an already complex code